### PR TITLE
[NDM][NDMII-2635] activate the autodiscovery automatically

### PR DESCRIPTION
**Pull request by jedupau**
### What does this PR do?

activate the autodiscovery automatically if at least one config is given under network_devices.snmp_listener

### CODE REVIEW

**Review by AlexandreYang — APPROVED** [review]
Nice!! 💯

**Review by clamoriniere — COMMENTED** [review]
Hi @jedupau 

the implementation might work, 
but have you looked at the `config/autodiscovery/env` Feature framework:
* https://github.com/DataDog/datadog-agent/blob/80c50f476edf2346bf0b13a08994879eecdcfd8e/pkg/config/env/environment_detection.go#L62
* we use it for container related features https://github.com/DataDog/datadog-agent/blob/80c50f476edf2346bf0b13a08994879eecdcfd8e/pkg/config/env/environment_containers.go

This FWK is generic and we can imaging to extended it to NDM feature.

File: pkg/config/autodiscovery/autodiscovery_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+	err := config.Datadog.ReadConfig(strings.NewReader(`
+network_devices:
+  snmp_listener:
+    configs:
+      - network: 127.0.0.1/30
+`))
+	assert.NoError(t, err)
+	_, configListeners := DiscoverComponentsFromConfig()
+	assert.Equal(t, 1, len(configListeners))
```
Comment (L3n41c) [inline pkg/config/autodiscovery/autodiscovery_test.go:27]:
```suggestion
	assert.Len(t, configListeners, 1)
```

File: pkg/config/autodiscovery/autodiscovery_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	assert.Equal(t, "snmp", configListeners[0].Name)
+
+	err = config.Datadog.ReadConfig(strings.NewReader(`
+network_devices:
+  snmp_listener:
+    configs:
+`))
+	assert.NoError(t, err)
+	_, configListeners = DiscoverComponentsFromConfig()
+	assert.Equal(t, 0, len(configListeners))
```
Comment (L3n41c) [inline pkg/config/autodiscovery/autodiscovery_test.go:37]:
```suggestion
	assert.Empty(t, configListeners)
```

File: pkg/config/autodiscovery/autodiscovery_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+	err = config.Datadog.ReadConfig(strings.NewReader(`
+snmp_listener:
+  configs:
+    - network: 127.0.0.1/30
+`))
+	assert.NoError(t, err)
+	_, configListeners = DiscoverComponentsFromConfig()
+	assert.Equal(t, 0, len(configListeners))
```
Comment (L3n41c) [inline pkg/config/autodiscovery/autodiscovery_test.go:46]:
```suggestion
	assert.Empty(t, configListeners)
```

**Review by L3n41c — APPROVED** [review]
I only left minor suggestions to use more specific testify assertion functions.
The main impact would be clearer error messages in case of failure.

File: pkg/config/autodiscovery/autodiscovery_test.go (hunk 1 above)
Comment (jedupau) [reply pkg/config/autodiscovery/autodiscovery_test.go]: thanks for the tip !

**Comment by TCheruy** [comment]
Hi @clamoriniere

Thank you for suggesting! We took a look a the env feature framework, but realised it might require some non-trivial work for us. Since we don't have plan on having other feature than SNMP in the autodiscovery, we'd like to go with this implementation as it's easier for us.
What we're looking to do with this PR is to remove the requirement for
```
listeners:
  - name: snmp
```
in datadog.yaml when any `snmp_listener` conf is given (since it's redundant)
 
 We can discuss this on slack if needed!

**Review by cswatt — APPROVED** [review]
Docs-approved

**Pull request merged**